persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -192,7 +194,8 @@ public ResultMessage processPrepared(
     }
 
     if (customPayload != null && customPayload.containsKey("token")) {
-      authorizeByToken(customPayload, statement);
+      // todo extract tenant_id and map to header
+      authorizeByToken(customPayload, statement, Collections.emptyMap());
```

Comment:
There is a problem with accessing tenant_id here (same for `StargateQueryHandler` for C* 4 and DSE).
In all places that I marked with ` // todo extract tenant_id and map to header` we need extract the tenant_id somehow.
In this context, we have access to `  QueryState queryState,` which has the `clientState`.
Unfortunately, the clientState does not carry the same info as `ClientInfo`. It has only `remoteAddress` and no public address(`sourceAddress`) (`queryState.getClientState().getRemoteAddress()`). 
This PR: https://github.com/stargate/stargate/pull/564/files propagates the tenant_id using `sourceAddress`.
@tjake Is there a way to inject that information to `queryState` as well?

Reply:
Each backend needs to have a ClientStateWithTenant subclass that is setup when the connection is created.

Reply:
yes, it is using the `ClientState` now. This comment got outdated, resolving now.